ignition-server/controllers/tokensecret_controller.go

```diff
@@ -145,6 +176,12 @@ func (r *TokenSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
+	// if a token is expired remove it
+	if isTokenExpired(log, tokenSecret.Annotations) {
+		err := r.processExpiredToken(ctx, tokenSecret)
+		return ctrl.Result{}, err
+	}
```

Comment:
technically if a token is not expired yet but already have the annotation, below it should `return ctrl.Result{RequeueAfter: min(now - IgnitionServerTokenExpirationTimestampAnnotation, ttl/2 - durationDeref(timeLived))}, nil` though that might make the logic to complex and without it we are eventually consistent anyways.

Reply:
That's what I was thinking it's nice and clean as is and ends up going through so I think it should be good!